botocore/serialize.py

```diff
@@ -786,20 +846,55 @@ def _serialize_type_boolean(self, xmlnode, params, shape, name):
         else:
             str_value = 'false'
         node.text = str_value
+        self._add_xml_namespace(shape, node)
 
     def _serialize_type_blob(self, xmlnode, params, shape, name):
         node = ElementTree.SubElement(xmlnode, name)
         node.text = self._get_base64(params)
+        self._add_xml_namespace(shape, node)
 
     def _serialize_type_timestamp(self, xmlnode, params, shape, name):
         node = ElementTree.SubElement(xmlnode, name)
-        node.text = self._convert_timestamp_to_str(
-            params, shape.serialization.get('timestampFormat')
+        node.text = str(
+            self._convert_timestamp_to_str(
+                params, shape.serialization.get('timestampFormat')
+            )
         )
+        self._add_xml_namespace(shape, node)
 
     def _default_serialize(self, xmlnode, params, shape, name):
         node = ElementTree.SubElement(xmlnode, name)
         node.text = str(params)
+        self._add_xml_namespace(shape, node)
+
+    def _serialize_content_type(self, serialized, shape, shape_members):
+        """Set Content-Type to application/xml for all structured bodies."""
+        if has_header('Content-Type', serialized['headers']):
+            return
+        payload = shape.serialization.get('payload')
+        if self._has_streaming_payload(payload, shape_members):
+            if shape_members[payload].type_name == 'string':
+                serialized['headers']['Content-Type'] = 'text/plain'
+            elif shape_members[payload].type_name == 'blob':
+                serialized['headers']['Content-Type'] = (
+                    'application/octet-stream'
+                )
+        else:
+            if serialized['body'] != b'':
+                serialized['headers']['Content-Type'] = 'application/xml'
+
+    def _add_xml_namespace(self, shape, structure_node):
+        if 'xmlNamespace' in shape.serialization:
```

Comment:
I need to take a closer look to confirm this, but after spending about ten minutes on it, I am wondering if there's a gap with the protocol tests here.  

Many services seem to be using `xmlNamespace` on their top level metadata, and I don't see a test for that.  Route53 and S3Control put `xmlNamespace` on `operations[<operationName>]['input']['xmlNamespace']`, and I don't see a test for that either

Reply:
Good call out. I'll make a note of this and we can potentially get these upstreamed into the smithy protocol tests if applicable. The xmlNamespace on the input structure makes sense and we handle that today. 

We currently don't do anything for xmlNamespace applied on the service and I'm honestly not sure what that's supposed to result in.

I don't think this is a blocker though. Lmk if you think differently